File: SDKs/googleapis.com/androiddeviceprovisioning/v1/python/src/sdk/models/shared/finddevicesbyownerresponse.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from . import device


@dataclass_json
@dataclass
class FindDevicesByOwnerResponse:
    devices: Optional[List[device.Device]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'devices' }})
    next_page_token: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'nextPageToken' }})
    total_size: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'totalSize' }})
    

File: SDKs/googleapis.com/androiddeviceprovisioning/v1/typescript/src/sdk/models/shared/finddevicesbyownerresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Device } from "./device";



// FindDevicesByOwnerResponse
/** 
 * Response containing found devices.
**/
export class FindDevicesByOwnerResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=devices", elemType: Device })
  devices?: Device[];

  @SpeakeasyMetadata({ data: "json, name=nextPageToken" })
  nextPageToken?: string;

  @SpeakeasyMetadata({ data: "json, name=totalSize" })
  totalSize?: number;
}
